src/OpenTelemetry.Resources.Gcp/GcpResourceDetector.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using Google.Api.Gax;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Resources.Gcp;
+
+/// <summary>
+/// Resource detector for Google Cloud Platform (GCP).
+/// </summary>
+public sealed class GcpResourceDetector : IResourceDetector
```

Comment:
nit: you can keep this internal. The extension method is public, and that should be good enough for users.

Reply:
Wasn't sure if that part of #1610 was going to be adopted 👍